### PR TITLE
added a new docker-arm profile to build docker containers in arm64 

### PR DESCRIPTION
**Pull request by sriki-cflt**
This change adds support to build docker images on `arm64` machines/laptops.

The following are the changes I have made

1. Added a new profile docker-arm that uses the [io.fabric8:docker-maven-plugin](https://github.com/fabric8io/docker-maven-plugin)
2. This plugin supports docker image building in `arm64` laptops.

_Ensuring Backward compatibility and minimum interference with existing plugin:_
1. Completely new build profile isolates the build process.
2. added additional property: `docker.imagePullPolicy` default in the common default this will not be used by existing plugin hence no interference.
3. the new plugin requires defaults for some of existing properties; added these as defaults in plugin scope so that they do not impact the existing plugin.

### CODE REVIEW

**Comment by janjwerner-confluent** [comment]
@sriki-cflt 
This scope of work will require review from DevProd as this affects the build pipeline more than security. I have requested their review. 
In the meantime - the downstream validation fails building images - that will need a resolution.

**Comment by sriki-cflt** [comment]
@janjwerner-confluent Thank you for point out the build failures. I have rerun the build it has passed.
https://github.com/confluentinc/common/pull/563/checks?check_run_id=18810208067
The failures were 403 for downloading while downloading artifacts - hence re run solved it.

**Comment by janjwerner-confluent** [comment]
great, looks good to me, can you please wait for someone from @confluentinc/devprod to sign of it as well.

**Review by janjwerner-confluent — APPROVED** [review]
looks good to me, it's just adding another build profile. 
@confluentinc/devprod 
this falls much closer to your expertise, can you please chime on it.

**Comment by rramakrishnan7** [comment]
@wadhwa1 can you approve changes from cp-release side.
@xli1996 comment on slack.

> The changes overall lgtm, but devprod is not codeowner to these two repos anymore. Seems security and cp-release team’s approval will grant the merge.

**Review by wadhwa1 — APPROVED** [review]
Thank you @sriki-cflt , lgtm
btw, I just noticed, that we have the same activation block for both profiles - `docker` and `docker-arm` , which is based on env var set, how will it choose which one to pick? 🤔 
`<activation>
    <property>
        <name>env.DOCKER_HOST</name>
    </property>
</activation>
`

**Review by rramakrishnan7 — APPROVED** [review]

**Pull request merged**
